pkg/logentry/stages/json_test.go

```diff
@@ -31,28 +42,51 @@ var testJSONLogLine = `
 	"level" : "WARN",
 	"nested" : {"child":"value"},
     "duration" : 125,
-	"message" : "this is a log line"
+	"message" : "this is a log line",
+	"extra": "{\"user\":\"marco\"}"
 }
 `
 
 func TestPipeline_JSON(t *testing.T) {
-	expected := map[string]interface{}{
-		"out":      "this is a log line",
-		"app":      "loki",
-		"nested":   "{\"child\":\"value\"}",
-		"duration": float64(125),
+	tests := map[string]struct {
+		config          string
+		entry           string
+		expectedExtract map[string]interface{}
+	}{
+		"successfully run a pipeline with 1 json stage without source": {
+			testJSONYamlSingleStageWithoutSource,
+			testJSONLogLine,
+			map[string]interface{}{
+				"out":      "this is a log line",
+				"app":      "loki",
+				"nested":   "{\"child\":\"value\"}",
+				"duration": float64(125),
+			},
+		},
+		"successfully run a pipeline with 2 json stages with source": {
+			testJSONYamlMultiStageWithSource,
+			testJSONLogLine,
+			map[string]interface{}{
+				"extra": "{\"user\":\"marco\"}",
+				"user":  "marco",
+			},
+		},
 	}
 
-	pl, err := NewPipeline(util.Logger, loadConfig(testJSONYaml), nil, prometheus.DefaultRegisterer)
-	if err != nil {
-		t.Fatal(err)
+	for testName, testData := range tests {
+		t.Run(testName, func(t *testing.T) {
```

Comment:
Can you set this up for this test to run in parallel?  Like the `TestJSONParser_Parse`